test/parallel_api/algorithm/alg.modifying.operations/transform_binary_zip.pass.cpp

```diff
@@ -0,0 +1,168 @@
+// -*- C++ -*-
+//===-- transform_binary.pass.cpp -----------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(execution)
+#include _PSTL_TEST_HEADER(algorithm)
+
+#include "support/utils.h"
+
+using namespace TestUtils;
+
+template <typename TInput1, typename TInput2, typename TOutput>
+class TheOperation
+{
+    TOutput val;
+
+public:
+
+    TheOperation(TOutput v) : val(v) {}
+
+    TOutput
+    operator()(const ::std::tuple<const TInput1&, const TInput1&>& x,
+               const ::std::tuple<const TInput2&, const TInput2&>& y) const
+    {
+        return TOutput(val + (::std::get<0>(x) + ::std::get<1>(x)) - (::std::get<0>(y) + ::std::get<1>(y)));
+    }
```

Comment:
From the PR description:
"Solved problem: binary transform could not work with zip_iterator as an output."   

I think the original motivation for this PR is to allow `zip_iterators` to be used as output sequences for binary transform operations.  It seems from this test that we are testing `zip_iterator` inputs, but not `zip_iterator` outputs.  Am I missing that somewhere?

Reply:
good catch!
yes, new test cases should cover  a proxy object like tuple<ref, ref> as an output value.